[jenkins] Moved data source from openshift to jenkins

mocenas opened the pull request:
Changed rourka to get data from jenkins instead from openshift.
Added possibility to run builds from rourka

mocenas committed:
[jenkins] Added connection to jenkins and loading data from it

mocenas committed:
[jenkins] Display data from jenkins on dashboard

mocenas committed:
[jenkins] Add parsing json from jenkins description field

mocenas committed:
[jenkins] Added running builds through jenkins

Ladicek commented on dashboard/src/main/java/com/github/ladicek/rourka/BuildResource.java:
```diff
@@ -0,0 +1,36 @@
+    @GET
+    @Produces("text/plain;charset=utf-8")
+    public String get(@PathParam("name") String buildName) throws Exception {
+        jenkinsBuildRunner = new JenkinsBuildRunner(new JenkinsRequestHandler(httpClient));
```
Shouldn't be a field, just a local variable.

Ladicek commented on dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsAuthorizedHttpClientProducer.java:
```diff
@@ -0,0 +1,60 @@
+
+
+        String authStr = jenkinsCredentials.getLogin() +":"+  jenkinsCredentials.getApiToken();
+        String encodedCredentials = DatatypeConverter.printBase64Binary(authStr.getBytes("utf-8"));
```
Use `java.util.Base64`.

Ladicek commented on dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsDataProvider.java:
```diff
@@ -0,0 +1,172 @@
+        }
+    }
+
+    private Boolean readBooleanOrNull(JsonObject object, String attribute)
```
These two methods have very similar names, but treat `null` values very differently.

Ladicek commented on dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsProperties.java:
```diff
@@ -0,0 +1,29 @@
+     */
+    static {
+        try {
+            ClassLoader loader = JenkinsProperties.class.getClassLoader();
```
This won't work. I will want to store the credentials into an OpenShift secret, which will be exposed either as environment variables or a file. Probably a file.

Ladicek commented on dashboard/src/main/java/com/github/ladicek/rourka/jenkins/JenkinsRequestHandler.java:
```diff
@@ -0,0 +1,70 @@
+    @JenkinsAuthorizedHttpClient
+    private CloseableHttpClient httpClient;
+
+    public JenkinsRequestHandler(HttpClient httpClient) throws Exception{
```
Why do you `@Inject` the HTTP client above, then?

Ladicek commented on dashboard/src/main/resources/jenkins.properties:
```diff
@@ -0,0 +1,5 @@
+jenkins.url=http://jenkins
+
+jenkins.login=xpaasqe-admin
+jenkins.apiToken=3bf6ab02b4928a54c043cbb14483199c
```
Don't publish secrets! Delete this file from the PR and if this is real token, contact someone to invalidate it right away.

Ladicek reviewed (changes requested):
Please fix the code style, too. (Opening braces always on the same line, spaces around assignment operator, etc.)

Also please cut the crap (`AbstractValueTransfer`???), especially when it comes to mutable objects. No JavaBeans.

Otherwise mostly look good. I will merge this when I'm finished with the refactoring -- the dashboard comes last, even though you were able to finish it first :-)

Ladicek commented:
Oh, one more thing. I don't recall if the code already deals with it or not, but there will be Jenkins jobs _without_ the JSON in description. Some kind of "support" jobs. That's okay and expected; these jobs should simply be skipped, they won't be reported in the dashboard.

Ladicek commented:
Actually, one other question: the original approach of authenticating to Jenkins using the service account's token worked just fine. Why did you change it?

mocenas commented:
I changed the authentication because the authenticating thought service account's token logged user in as a service account -> which has read only access so it cannot run builds. 
I was unable to change priviledge to a service account so I changed auth. to be able to run builds

mocenas committed:
Delete jenkins.properties

Ladicek commented:
Hmm, makes sense. I think it should be doable, but let's keep your change for now. I'll investigate later.

mocenas commented:
Well am sorry. I already investigated it deeper and found out that it is doable and I done it. So in my new version it is again done by token authentication.

Ladicek commented:
Nice, thanks! :-)

mocenas committed:
[jenkins] Fixed code style and remarks.

mocenas committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	dashboard/src/main/resources/jenkins.properties

mocenas committed:
[jenkins] Indentation changed to spaces

mocenas commented:
Ok I changed indentation from tabs to spaces

Ladicek commented:
I've merged this locally and did a significant refactoring. Closing this PR, but none of this is actually dropped. Thanks!
